File: momhelp/src/main/java/com/momhelp/service/ImageDetectionService.java
package com.momhelp.service;

import org.springframework.web.multipart.MultipartFile;
import java.util.List;
import java.util.Map;

public interface ImageDetectionService {

	// Detect vegetables from uploaded image
	List<Map<String, Object>> detectVegetables(MultipartFile image) throws Exception;

	// Validate image file
	boolean validateImage(MultipartFile image);

	// Get supported formats
	String[] getSupportedFormats();
}

File: momhelp-frontend/momhelp-frontend/src/services/imageDetectionService.js
import api from './api';

const imageDetectionService = {
  // Detect vegetables from image
  detectVegetables: (imageFile) => {
    const formData = new FormData();
    formData.append('image', imageFile);

    return api.post('/image/detect-vegetables', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    });
  },

  // Get supported formats
  getSupportedFormats: () => {
    return api.get('/image/supported-formats');
  },

  // Test endpoint
  testService: () => {
    return api.get('/image/test');
  }
};

export default imageDetectionService;